pyOCD/flash/flash_builder.py

```diff
@@ -422,3 +504,106 @@ def _page_erase_program(self, progress_cb = _stub_progress):
         logging.debug("Actual page erase count: %i", actual_page_erase_count)
 
         return FLASH_PAGE_ERASE
+
+    def _scan_pages_for_same(self, progress_cb = _stub_progress):
+        """
+        Program by performing sector erases.
+        """
+        progress = 0
+        count = 0
+        same_count = 0
+
+        for page in self.page_list:
+            # Read page data if unknown - after this page.same will be True or False
+            if page.same is None:
+                data = self.flash.target.readBlockMemoryUnaligned8(page.addr, len(page.data))
+                page.same = _same(page.data, data)
+                progress += page.getVerifyWeight()
+                count += 1
+                if page.same:
+                    same_count += 1
+
+                # Update progress
+                progress_cb(float(progress) / float(self.page_erase_weight))
+        logging.debug("scanned %d pages for equality (%d same)", count, same_count)
```

Comment:
If debug logging is on and only a few pages needed to be reprogrammed this will print after the progress bar making the progress bar appear twice:

```
DEBUG:root:Chip erase weight 37.320000, Page erase weight 15.065200
DEBUG:root:Using double buffer page erase program
[================    ]  83%DEBUG:root:scanned 245 pages for equality (245 same)
[====================] 100%
DEBUG:root:Estimated page erase count: 11
DEBUG:root:Actual page erase count: 11
```

This is pretty minor so it might not be worth fixing, but I figured I would mention it.  The estimated vs real page same count is printed later, so this statement might not be needed.